internal/modules/compass/models.go

```diff
@@ -0,0 +1,14 @@
+package compass
+
+import "encoding/json"
+
+type CompassEvent struct {
+	Time           string          `json:"time"`
```

Comment:
you can make this time.Time object and the json marshaling will handle the parsing for you.
```suggestion
	Time           time.Time          `json:"time"`
```